pd/src/components/staking.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use penumbra_chain::params::ChainParams;
+use penumbra_stake::Epoch;
+use penumbra_transaction::Transaction;
+use tendermint::abci;
+
+use super::{Component, Overlay};
+use crate::{components::validator_set::BlockChanges, genesis, WriteOverlayExt};
+
+// Stub component
+pub struct Staking {
+    overlay: Overlay,
+
+    // TODO: this shouldn't be here -- we should grab this from a common location
+    // in the JMT, i.e. genesis/AppState but that's not available until #511 lands.
+    // since consensus can't yet change chain params on the fly, it's not an issue,
+    // but it definitely will be in the future!
+    chain_params: Option<ChainParams>,
```

Comment:
Non-blocking:

Does this need to be here (i.e., in-memory, as a field on `Staking`) at all?  It looks like the only place it's read from is when getting the epoch duration.

But alternatively, we could have the epoch duration written into the JMT somewhere and have an `epoch_duration` method on `Staking` that would pull it out of the `Overlay` whenever it was needed.

There's a question of how we might want to organize code for reading / writing data that would be used by multiple components (e.g., the epoch duration).  We should break out discussion on this, but I think making another extension trait could be a good option.

Reply:
Yeah, I'm removing this as part of the implementation and only fetching it from the JMT (and for now it'll be set as part of the `App` as discussed)